crates/foundry-ui/src/message.rs

```diff
@@ -0,0 +1,31 @@
+use serde::Serialize;
+
+/// A typed object that can be either printed as a human-readable message or serialized as JSON.
+pub trait Message {
+    /// Return textual (human) representation of this message.
+    ///
+    /// Default implementation returns empty string, making [`Ui`] skip printing this message.
+    fn text(&self) -> String
+    where
+        Self: Sized + Serialize;
+
+    /// Return JSON representation of this message.
+    fn json(&self) -> String
+    where
+        Self: Sized + Serialize,
+    {
+        serde_json::to_string(self).unwrap_or_else(|_| "Invalid JSON".to_string())
+    }
+}
+
+impl Message for &str {
+    fn text(&self) -> String {
+        (*self).to_string()
+    }
+}
+
+impl Message for String {
+    fn text(&self) -> String {
+        self.to_string()
+    }
+}
```

Comment:
How does that play with the default conversion to json? The user will just get the encoded string?

Not sure how we can enforce this on the type level (or if), but for json output maybe it makes sense to return objects always? 

What I mean is that maybe there is a reason for the output to always be of the format

```
SomeObject { 
  "key1": "value1"
}
```

and with these default impls the user will get just `"string text"`.

cc @kkawula @ksew1